runtime/near-wallet-contract/implementation/wallet-contract/src/eth_emulation.rs

```diff
@@ -0,0 +1,87 @@
+//! This module contains logic for emulating Ethereum standards with the
+//! corresponding Near actions. For now only the ERC-20 standard is supported
+//! (which corresponds to Near's NEP-141).
+
+use crate::{
+    error::{Error, UserError},
+    ethabi_utils,
+    types::{Action, ExecutionContext},
+};
+use aurora_engine_transactions::NormalizedEthTransaction;
+use ethabi::{Address, ParamType};
+use near_sdk::AccountId;
+
+const FIVE_TERA_GAS: u64 = near_sdk::Gas::from_tgas(5).as_gas();
+
+pub const ERC20_BALANCE_OF_SELECTOR: &[u8] = &[0x70, 0xa0, 0x82, 0x31];
+const ERC20_BALANCE_OF_SIGNATURE: [ParamType; 1] = [ParamType::Address];
+
+pub const ERC20_TRANSFER_SELECTOR: &[u8] = &[0xa9, 0x05, 0x9c, 0xbb];
+const ERC20_TRANSFER_SIGNATURE: [ParamType; 2] = [
+    ParamType::Address,   // to
+    ParamType::Uint(256), // value
+];
+
+pub fn try_emulation(
+    target: &AccountId,
+    tx: &NormalizedEthTransaction,
+    context: &ExecutionContext,
+) -> Result<Action, Error> {
+    if tx.data.len() < 4 {
+        return Err(Error::User(UserError::InvalidAbiEncodedData));
+    }
+    let suffix = context
+        .current_account_id
+        .as_str()
+        .find('.')
+        .map(|index| &context.current_account_id.as_str()[index..])
+        .unwrap_or("");
+    match &tx.data[0..4] {
+        ERC20_BALANCE_OF_SELECTOR => {
+            let (address,): (Address,) =
+                ethabi_utils::abi_decode(&ERC20_BALANCE_OF_SIGNATURE, &tx.data[4..])?;
+            let args = format!(
+                r#"{{"account_id": "0x{}{}"}}"#,
```

Comment:
In production there will not be any suffix because eth-implicit accounts are top-level. But for testing (eg in `near-workspaces`) I assume that all wallet contracts are deployed as sub-accounts of the same namespace, hence they all share the same suffix.

Reply:
Could we add a short comment that the suffix stuff is for testing purposes?

Reply:
Done in https://github.com/near/nearcore/pull/10850/commits/ee16f2bf75aabea09ecad4ccf1a44d6950ffd26b